Owncloud iOs Client/InstantUpload/InstantUpload.m

```diff
@@ -0,0 +1,242 @@
+                            ACTIVE_USER.timestampInstantUpload = assetDateCreated;
+                            [ManageAppSettingsDB updateTimestampInstantUpload:assetDateCreated];
+                        }
+                    }
+                    
+                    AppDelegate *app = (AppDelegate *)[[UIApplication sharedApplication] delegate];
+                    
+                    if (app.prepareFiles == nil) {
+                        app.prepareFiles = [[PrepareFilesToUpload alloc] init];
+                        app.prepareFiles.listOfFilesToUpload = [[NSMutableArray alloc] init];
+                        app.prepareFiles.listOfAssetsToUpload = [[NSMutableArray alloc] init];
+                        app.prepareFiles.arrayOfRemoteurl = [[NSMutableArray alloc] init];
+                        app.prepareFiles.listOfUploadOfflineToGenerateSQL = [[NSMutableArray alloc] init];
+                    }
+                    
+                    app.prepareFiles.delegate = app;
+                    app.prepareFiles.counterUploadFiles = 0;
+                    app.uploadTask = [[UIApplication sharedApplication] beginBackgroundTaskWithExpirationHandler:^{
+                        // If you’re worried about exceeding 10 minutes, handle it here
+                        
+                    }];
+                    
+                    [app.prepareFiles addAssetsToUpload:newPhotos andRemoteFolder:k_path_instant_upload];
+                }
+                
+            } else {
+                [self setEnabled:NO];
+                [self showPhotoLibraryAccessPermissionDeniedAlert];
+                [self.delegate instantUploadPermissionLostOrDenied];
+            }
+        }
+    }
+}
+
+#pragma mark - Background Instant Upload
+
+- (BOOL) isBackgroundLocationUpdatesPermissionGranted {
+    if ([CLLocationManager locationServicesEnabled]) {
+        if ([CLLocationManager authorizationStatus] == kCLAuthorizationStatusAuthorizedAlways) {
+            return YES;
+        }
+    }
+    return NO;
+}
+
+#pragma mark CLLocationManagerDelegate methods
+
+-(void)locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArray *)locations {
+    [self attemptUpload];
+}
+
+-(void)locationManager:(CLLocationManager *)manager didChangeAuthorizationStatus:(CLAuthorizationStatus)status {
+    if ([self backgroundInstantUploadEnabled]) {
+        if (status == kCLAuthorizationStatusAuthorizedAlways) {
+            [self.locationManager startMonitoringSignificantLocationChanges];
+        } else {
+            [self setBackgroundInstantUploadEnabled:NO];
+            [self showLocationAccessAlwaysPermissionDeniedAlert];
+            [self.delegate backgroundInstantUploadPermissionLostOrDenied];
+        }
+    }
+}
+
+#pragma mark PHPhotoLibraryChangeObserver methods
+
+- (void)photoLibraryDidChange:(PHChange *)changeInstance {
+    [self attemptUpload];
+}
+
+#pragma mark - Utility
+
+- (void) showPhotoLibraryAccessPermissionDeniedAlert {
```

Comment:
Well, since you are using title and body use this method that you are not using
```
(void) showAlertViewWithTitle:(NSString *)title body:(NSString *)body{ 
``` 
instead of the other two
```
showLocationAccessAlwaysPermissionDeniedAlert
showLocationAccessAlwaysPermissionDeniedAlert
```